src/Link.php

```diff
@@ -86,12 +91,7 @@ public static function create(string $title, \DateTimeInterface $from, \DateTime
      */
     public static function createAllDay(string $title, \DateTimeInterface $fromDate, int $numberOfDays = 1): self
     {
-        // In cases where the from date is not UTC, make sure it's UTC, size all day events are floating and non UTC dates cause bugs in the generators
-        if ($fromDate->getTimezone() !== new \DateTimeZone('UTC')) {
-            $fromDate = \DateTime::createFromFormat('Y-m-d', $fromDate->format('Y-m-d'));
-        }
-
-        $from = (clone $fromDate)->modify('midnight');
+        $from = (clone $fromDate);
```

Comment:
With my old code, we're removing the hours/days portion. I guess this probably wasn't required and it only was done for non UTC dates.